frontend/hub/api/utils.tsx

```diff
@@ -0,0 +1,222 @@
+}
+
+export function idKeyFn(item: { id: number | string }) {
+  return item.id;
+}
+
+export function collectionKeyFn(item: {
+  collection_version: { pulp_href: string };
+  repository: { name: string };
+}) {
+  return item.collection_version.pulp_href + '_' + item.repository.name;
+}
+
+export function appendTrailingSlash(url: string) {
+  return url.endsWith('/') ? url : url + '/';
+}
+
+export async function hubAPIDelete<T extends object>(url: string, signal?: AbortSignal) {
+  try {
+    const { response, statusCode } = await deleteHubRequest<T>(url, signal);
+    if (statusCode === 202) {
+      await parseTaskResponse(response as TaskResponse, signal);
+    }
+  } catch (error) {
+    throw new Error('Error deleting item');
+  }
+}
+
+export async function hubAPIPatch<T extends object, RequestBody = unknown>(
+  url: string,
+  data: RequestBody,
+  signal?: AbortSignal
+) {
+  try {
+    const { response, statusCode } = await patchHubRequest<T>(url, data, signal);
+    if (statusCode === 202) {
+      await parseTaskResponse(response as TaskResponse, signal);
+    }
+  } catch (error) {
+    throw new Error('Error updating item');
+  }
+}
+
+export async function hubAPIPut<T extends object, RequestBody = unknown>(
+  url: string,
+  data: RequestBody,
+  signal?: AbortSignal
+) {
+  try {
+    const { response, statusCode } = await putHubRequest<T>(url, data, signal);
+    if (statusCode === 202) {
+      await parseTaskResponse(response as TaskResponse, signal);
+    }
+  } catch (error) {
+    throw new Error('Error updating item');
+  }
+}
+
+async function parseTaskResponse(response: TaskResponse | undefined, signal?: AbortSignal) {
+  if (response?.task) {
+    const taskHref = parsePulpIDFromURL(response.task);
+    if (taskHref) {
+      await waitForTask(taskHref, signal);
+    }
+  }
+}
+
+/**
+ * Waits for a task to complete, given its href.
+ *
+ * Certain API actions return a response with a status code 202, containing a task reference
+ * like `{ task: task_href }`. This function is used to wait for the referenced task to complete
+ * successfully or fail with an error.
+ *
+ * @param {string | null} taskHref - The href of the task to wait for.
+ * @param {AbortSignal} [signal] - An optional AbortSignal for aborting the waiting.
+ * @param {number} [minDelay=5] - The initial delay between retries, in milliseconds.
+ * @param {number} [maxDelay=1000] - The maximum delay between retries, in milliseconds.
+ * @param {number} [retries=20] - The maximum number of retries.
+ *
+ * @throws {Error} Throws an error if the task href is invalid or if the task did not finish within the specified retries.
+ * @throws {Error} Throws an error with task error description if the task fails with an error.
+ * @throws {Error} Throws an error if the task is not found (status code 404).
+ */
+export async function waitForTask(
+  taskHref: string | null,
+  signal?: AbortSignal,
+  minDelay = 5,
+  maxDelay = 1000,
+  retries = 20
```

Comment:
5ms is too low for minDelay, and so is 1s for maxDelay

also, this will wait 20s max, this needs to keep trying for at least several minutes, without overloading the server by too many requests

Reply:
@himdel Thanks for the review, I will update as suggested.